src/components/ImgUpload/index.jsx

```diff
@@ -77,45 +77,54 @@ export default function ImgUpload(props) {
     setImagePreviews(images);
   };
 
-  const savePost = (imageUrl = "") => {
-    db.collection("posts")
-      .add({
+  const savePost = async (imageUrl = "") => {
+    try {
+      const postRef = await db.collection("posts").add({
         timestamp: firebase.firestore.FieldValue.serverTimestamp(),
         caption: caption,
         imageUrl,
         username: props.user.displayName,
         avatar: props.user.photoURL,
         likecount: [],
-      })
-      .then(() => {
-        playSuccessSound();
-        enqueueSnackbar("Post uploaded successfully!", {
-          variant: "success",
-        });
-        setProgress(0);
-        setCaption("");
-        setImage(null);
-        if (imgInput.current) {
-          imgInput.current.value = null;
-        }
+        uid: auth?.currentUser?.uid,
+      });
 
-        if (props.onUploadComplete) {
-          props.onUploadComplete();
-        }
-      })
-      .catch((err) => {
-        playErrorSound();
-        enqueueSnackbar(err.message, {
-          variant: "error",
+      const postId = postRef.id; // Store post ID in a separate variable
+
+      await db
+        .collection("users")
+        .doc(props.user.uid)
+        .update({
```

Comment:
if the document does not exists, it'll give error, use some property like upsert=true of mongodb

Reply:
set method will fix perhaps, right?

Reply:
old users should also be able to work on the app without any error

Reply:
That's Done.